commands/rootfs.go

```diff
@@ -78,12 +79,34 @@ func extractLayer(dest string, layer v1.Layer, bar *mpb.Bar, chown bool) error {
 		return err
 	}
 
-	gr, err := gzip.NewReader(bar.ProxyReader(r))
+	digest, err := layer.Digest()
 	if err != nil {
 		return err
 	}
 
-	tr := tar.NewReader(gr)
+	log := logrus.WithFields(logrus.Fields{
+		"Layer": digest.Hex[0:12],
+	})
+
+	var cr io.Reader
+	var crc func() error
+
+	gr, err := gzip.NewReader(bar.ProxyReader(r))
+	crc = func() error { return gr.Close() }
+	cr = gr
+	if err != nil {
+		log.Debugf("failed to decompress layer with gzip, trying zstd instead")
+
+		zr, err := zstd.NewReader(bar.ProxyReader(r))
+		crc = func() error { zr.Close(); return nil }
+		cr = zr
+		if err != nil {
+			log.Errorf("failed to decompress layer with gzip / zstd")
+			return err
+		}
+	}
```

Comment:
Instead of blindly falling back to trying zstd, why not inspect the `layer.MediaType()` for `types.OCILayer`/`types.OCILayerZStd`? Will make it easier in the future as well if we need to support more compression methods.